jvm-packages/xgboost4j/pom.xml

```diff
@@ -13,6 +13,16 @@
     <packaging>jar</packaging>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-hdfs</artifactId>
+            <version>2.7.3</version>
```

Comment:
This may be problematic in our (Criteo) environment and in general too.
The problem is that currently `xgboost-spark` depends on Spark 2.4.3 (https://github.com/dmlc/xgboost/blob/551db2269f037337dc4ef464e8e278c5f1eca607/jvm-packages/pom.xml#L37)
and it depends on Hadoop 2.6.5
So I suggest two things
1. change it to 2.6.5 unless you really need 2.7.3
2. Move this to a parent pom as a property and use it here

Also note you are including this as a compile dependency which may be not ideal at least in some environments but here I am no expert. I am familiar with our Hadoop|Yarn cluster only and our way of managing Spark jobs which is a bit weird :)

Reply:
I changed to provided and keep 2.7.3 , the reason is that 2.7.3 is kind of standard support in spark, if you look at spark, it only provides pre-build for 2.7

Reply:
I forgot that you target Spark 3.0 in this release which supports only Hadoop 2.7
Making dependency `provided` should solve Criteo's problem